packages/grafana-ui/src/types/plugin.ts

```diff
@@ -112,14 +112,14 @@ export class GrafanaPlugin<T extends PluginMeta = PluginMeta> {
   angularConfigCtrl?: any;
 
   // Show configuration tabs on the plugin page
-  configTabs?: Array<PluginConfigTab<T>>;
+  configPages?: Array<PluginConfigPage<GrafanaPlugin>>;
 
   // Tabs on the plugin page
-  addConfigTab(tab: PluginConfigTab<T>) {
-    if (!this.configTabs) {
-      this.configTabs = [];
+  addConfigPage(tab: PluginConfigPage<GrafanaPlugin>) {
```

Comment:
@dprokop  -- I think I got the type here kinda wrong.  This forces it to be ` PluginConfigPage<GrafanaPlugin>`, but not `PluginConfigPage<AppPlugin>` or `PluginConfigPage<DataSourcePlugin>`

Any thoughts on how to make that work?

Essentially I want to capture the class that was created.  I tried, `PluginConfigPage<GrafanaPlugin<T>>` but it fails with:
```
Type 'GrafanaPlugin<T>' does not satisfy the constraint 'GrafanaPlugin<PluginMeta>'.
  Type 'PluginMeta' is not assignable to type 'T'.ts(2344)
```